frontend-ssr/components/Scroll/Scroll.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable @typescript-eslint/no-empty-interface */
+import React from 'react';
+import BScroll from 'better-scroll';
+import throttle from 'lodash/throttle';
+
+import styles from './Scroll.module.scss';
+
+const INFINITE_SCROLL_DISTANCE = 100;
+
+interface State {}
+
+interface Props {
+  hasInfiniteScroll: boolean;
+  /** we use the property isReversed here so we can listen to the bottom or the
+   * top of the page */
+  isReversed?: boolean;
+  /** fetch data when we reach the top of the page, maybe we will need to do that in
+   *  some components */
+  onStartReach?: () => Promise<void>;
+  /** fetch data when we reach the end of the page */
+  onEndReach: () => Promise<void>;
+  preloadDistance: number;
+  className?: string;
+}
+class Scroll extends React.Component<Props, State> {
+  forceStop: Record<string, unknown>;
+
+  scrollRef: React.RefObject<HTMLDivElement>;
+
+  bScroll: any;
+
+  constructor(props) {
+    super(props);
+    this.forceStop = {};
+    this.scrollRef = React.createRef();
+    this._checkStartReach = throttle(this._checkStartReach.bind(this), 50);
+    this._checkEndReach = throttle(this._checkEndReach.bind(this), 50);
+  }
+
+  componentDidMount() {
+    this.bScroll = new BScroll(this.scrollRef.current, {
+      scrollbar: true,
+      eventPassthrough: 'horizontal',
+      mouseWheel: true,
+      click: true,
+    });
+    this._setupScrollEvents();
+  }
+
+  componentDidUpdate(prevProps) {
+    let { hasInfiniteScroll, isReversed } = this.props;
+
+    if (prevProps.hasInfiniteScroll === hasInfiniteScroll) {
+      return;
+    }
+    if (hasInfiniteScroll) {
+      if (isReversed) {
+        this._disposeStartReachEvents();
+        this.forceStop.start = false;
+        this._setupStartReachEvents();
+      } else {
+        this._disposeEndReachEvents();
+        this.forceStop.end = false;
+        this._setupEndReachEvents();
+      }
+    } else if (isReversed) {
+      this.forceStop.start = true;
+      this._disposeStartReachEvents();
+    } else {
+      this.forceStop.end = true;
+      this._disposeEndReachEvents();
+    }
+  }
+
+  componentWillUnmount() {
+    this._disposeScrollEvents();
+  }
+
+  getPosition() {
+    return {
+      x: this.bScroll.x,
+      y: this.bScroll.y,
+      maxX: this.bScroll.maxScrollX,
+      maxY: this.bScroll.maxScrollY,
+    };
+  }
+
+  _disposeScrollEvents() {
+    this._disposeStartReachEvents();
+    this._disposeEndReachEvents();
+  }
+
+  _setupScrollEvents() {
```

Comment:
Let's rename all functions starting with `_` by removing the `_`. Since in DevDrive I haven't got TypeScript - I used this notation to indicate it's supposed to be a private function. But here we can actually use the `private` descriptor.